### PR TITLE
[5.2] Let use custom Application in Composer scripts

### PR DESCRIPTION
**Pull request by vlakoff**
Allow to use custom Application by extending `ComposerScripts::createApplication` and using the extended `ComposerScripts` class in `composer.json`.

### CODE REVIEW

File: src/Illuminate/Foundation/ComposerScripts.php [hunk 1]
```diff
@@ -49,4 +49,15 @@ private static function clearCompiled()
             @unlink($servicesPath);
         }
     }
+
+    /**
+     * Creates a Laravel application instance.
+     *
+     * @param  string  $basePath
+     * @return \Illuminate\Foundation\Application
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/ComposerScripts.php]: guess this should really be the contract here

Comment (vlakoff) [inline src/Illuminate/Foundation/ComposerScripts.php]: updated.

**Comment by vlakoff** [comment]
Refs #12827 and pings @barryvdh :)

**Comment by taylorotwell** [comment]
Just define your own Composer Scripts class in that case.

**Comment by vlakoff** [comment]
Maybe at least change `clearCompiled` visibility to `protected`?

**Comment by crynobone** [comment]
> Just define your own Composer Scripts class in that case.

Agreed with @vlakoff, having `private static function clearCompiled()` would mean any project extending the default `Illuminate\Foundation\Application` need to write separate implementation of Composer Scripts instead of extending `Illuminate\Foundation\ComposerScripts`.

screen shot 2016-03-26 at 3 15 07 pm

**Comment by barryvdh** [comment]
Then submit a PR and see if it gets accepted.

**Comment by crynobone** [comment]
This IS a good PR IMHO.

**Comment by barryvdh** [comment]
Oh sorry, mixed this with my original PR. This works for me too, but not my call..

**Pull request closed without merging**